uitools/cpp/Esri/ArcGISRuntime/Toolkit/BasemapGalleryController.cpp

```diff
@@ -508,16 +508,38 @@ namespace Toolkit {
    */
   void BasemapGalleryController::setCurrentBasemap(Basemap* basemap)
   {
-    if (basemap == m_currentBasemap)
-      return;
-
-    m_currentBasemap = basemap;
-    emit currentBasemapChanged();
+      auto connection = new QMetaObject::Connection();
+
+      auto apply = [basemap, this, connection](Error e){
+        if(e.isEmpty()){
+            if (basemap == m_currentBasemap)
+                  return;
+            if(!basemapMatchesCurrentSpatialReference(basemap)){
+                //force redraw for all the lsitview items.
+                emit m_gallery->dataChanged(m_gallery->index(0), m_gallery->index(std::max(m_gallery->rowCount() - 1, 0)));
+                return;
+            }
+            m_currentBasemap = basemap;
+            emit currentBasemapChanged();
 
-    if (m_geoModel && m_geoModel->basemap() != m_currentBasemap)
-    {
-      m_geoModel->setBasemap(m_currentBasemap);
-    }
+            if (m_geoModel && m_geoModel->basemap() != m_currentBasemap)
+            {
+              m_geoModel->setBasemap(m_currentBasemap);
+            }
+            disconnect(*connection);
+        }
+        else {
+            qDebug() << "problem in loading the layer";
+        }
+        delete connection;
+      };
+      if(basemap->baseLayers()->first()->loadStatus() != LoadStatus::Loaded){
+          connect(basemap->baseLayers()->first(), &Layer::doneLoading, this, apply);
+          basemap->baseLayers()->first()->load();
+      }
+      else {
+          apply(Error());
```

Comment:
💭  I'd maybe go for `Error{}` instead. Does this work on all compilers? Looks like a potential most-vexing parse problem.